language/move-bytecode-verifier/src/absint.rs

```diff
@@ -101,18 +81,9 @@ pub trait AbstractInterpreter: TransferFunctions {
             };
 
             let pre_state = &block_invariant.pre;
-            let post_state = match self.execute_block(block_id, pre_state, function_view) {
-                Err(e) => {
-                    block_invariant.post = BlockPostcondition::Error(e);
-                    // Stop analyzing after the first error occurred, to avoid the risk of
-                    // crashes as followup errors.
-                    break;
-                }
-                Ok(s) => {
-                    block_invariant.post = BlockPostcondition::Success;
-                    s
-                }
-            };
+            // Note: this will stop analysis after the first error occurs, to avoid the risk of
+            // subsequent crashes
+            let post_state = self.execute_block(block_id, pre_state, function_view)?;
```

Comment:
This is the meat of the PR--all of the other changes are in service of being able to use a `Result` here.